Edited the code to use user registration  and added a post schema for creating post contents.

rosesharonanchi opened the pull request.

rosesharonanchi committed:
Completed API handler for users and posts

rosesharonanchi committed:
Updated the apis

TATA-THECLAIRE reviewed (approved):
- Proper bcrypt usage, Clean routing, Constructor pattern (NewAPIHandler),Clear separation of routes.
- Missing authentication layer:  You implemented registration and login, but the rest of the routes do not use any Because of this, anyone can: delete any post, modify any post, create posts on behalf of any user

VERNA990 reviewed (commented):
I really like that you encrypt the password before storing it in the database, great attention to security! Overall, everything looks solid.

I did notice a few points that could be improved:

There seem to be some unnecessary statements in your queries. For example, using LIMIT 1 isn’t needed if your query definition already ensures that only one row is returned.

Following some best practices could make the code even better, such as:
- Creating a separate branch for code improvements while keeping the main branch for finalized code.
- Ensuring that the DELETE method does not include a request body, as it’s not standard practice.

rosesharonanchi committed:
Added request struct for functions this enforces seperation of concerns

rosesharonanchi committed:
Added a lint checker

rosesharonanchi committed:
Create GitHub Actions workflow for CI/CD

Add CI/CD workflow for Go projects with build, test, linter, and dependency review jobs.

rosesharonanchi committed:
Create CI/CD workflow for Go projects

Add CI/CD workflow for Go projects with build, test, linter, and dependency review jobs.

rosesharonanchi committed:
Added govulncheck

rosesharonanchi committed:
added a test and comment for no lint err check

rosesharonanchi committed:
Fix checkout action reference in workflow

rosesharonanchi committed:
updated the go.mod to fix govul check

rosesharonanchi committed:
update to my branch

rosesharonanchi committed:
Update golangci-lint action to version 9

rosesharonanchi committed:
update

rosesharonanchi committed:
Merge pull request #3 from rosesharonanchi/add_linting

update on migration file

rosesharonanchi committed:
Add checkout step to govulncheck job

rosesharonanchi committed:
Fix YAML syntax and update Go version references

rosesharonanchi committed:
deleted a duplicate yml file

rosesharonanchi committed:
Merge branch 'main' of https://github.com/rosesharonanchi/sqlc-example-api

rosesharonanchi committed:
Refactor GitHub Actions workflow for consistency
